libutils/atexit.c

```diff
@@ -71,7 +71,11 @@ void RegisterAtExitFunction(AtExitFn fn)
     atexit_functions = p;
 
     pthread_mutex_unlock(&atexit_functions_mutex);
-#endif
-
+/*
+    Don't register atexit() functions on windows due to race conditions 
+    around lock cleanup.
+*/
+#else 
     atexit(fn);
+#endif
```

Comment:
Since this change is in *libutils* I think we need to make sure all the tools use the ``SafeExit()`` mechanism, right? Otherwise the functions registered here are never run, right?

Reply:
Yes. I will look for all use of RegisterAtExitFunction() and ensure that they also use SafeExit(). Probably move SafeExit to atexit.h/c for sharing purposes right!?